src/pybind/mgr/object_format.py

```diff
@@ -0,0 +1,272 @@
+# object_format.py provides types and functions for working with
+# requested output formats such as JSON, YAML, etc.
+
+import enum
+import errno
+import json
+
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+try:
+    from typing import Protocol  # Protocol was added in Python 3.8
+except ImportError:
+
+    class Protocol:  # type: ignore
+        pass
+
+
+import yaml
+
+from mgr_module import HandlerFuncType
+
+
+class Format(enum.Enum):
+    plain = "plain"
+    json = "json"
+    json_pretty = "json-pretty"
+    yaml = "yaml"
+    xml_pretty = "xml-pretty"
+    xml = "xml"
+
+
+def _obj_to_json(obj: Any) -> Any:
+    tj = getattr(obj, "to_json", None)
+    if tj:
+        return tj()
+    return obj
+
+
+def _obj_to_yaml(obj: Any) -> Any:
```

Comment:
What this method is really returning? an **str** a **dict** 
.. I find it confusing to figure out what we would get as an output specially with the "fallback" mechanism to json in case Yaml conversion is not possible!

Reply:
Perhaps the names are confusing... I'll think about renaming them. In short, if the object has it's own `yaml_representer` method it returns the object, otherwise it runs the _obj_to_json function which in turn may return the original object or the result of that object's to_json method.

I'll note that this is "just" the existing logic, moved to a new location, and broken up into smaller functions.  If there's a consensus that I *should not* attempt to reuse the exsiting json/yaml serialization from the orchestrator module I can certainly do my own thing. I could probably make it simpler. However then the new method would be incompatible with the existing approach... so :man_shrugging: